### ISSUE TITLE
Q: What about using VizBox as alternative HRI ?

### ISSUE THREAD

**Issue by cdussieux** [RoboCupAtHome/RuleBook#577]
Hi,

What about merging efforts of user report (explainability) and asr bypassing into the same component : VizBox ?
It looks like it makes sense from an architecture point of view, as those two functionalities are basically bound to the same stuff.

In a word, we'd like to add some buttons to VizBox, and use it as a HRI. Didn't dive deep in VizBox yet, it is a work to be done on our side.
I wanted to ping you to see how well that would fit in the whole VizBox idea.
Btw would it be an issue for "Alternative HRIinterfaces (see Section 3.8.3) must be previously approved by the TC during the RobotInspection (see Section 4.4)." ?

Thanks
cdussieux

**Comment by LoyVanBeek** [comment]
The Vizbox is able to do this. The server hast http POST interface that is meant to accept sentences from the user. Not sure anymore how that is used in the UI directly, but I'm quite open to just have a text field in the bottom where you can type your response.
What the robot says can also be displayed already

**Comment by cdussieux** [comment]
I was more thinking of adding a whole button panel, with 3 or 4 buttons, relevant regarding to the current state of the robot. This functionality would rely on server http POST interface.
e.g. a button that could trigger state machine transition, from a user-friendly point of view.
And fulfill role expected of alternative HRI.

Please correct me if i'm wrong, but i do not see the point of an alternative HRI independent from VizBox, as its use is already expected.
(That said, it could be answered that the main goal of alternative HRI is rather about design and user friendliness.)

My question is more like, what would be the reason not to use VizBox as alternative HRI ?
Thanks

**Comment by LoyVanBeek** [comment]
You are free to do that and present that as HRI alternative!

**Comment by kyordhel** [comment]
I think these technicalities would find a better place as an issue in the [Vizbox repo](https://github.com/LoyVanBeek/vizbox)

> What would be the reason not to use VizBox as alternative HRI

# TL;DR: Please use Vizbox as alternate HRI interface

Other than current Vizbox limitations for the purpose, I can't find a reason for not using it. In fact I believe the initiative would greatly benefit the three leagues with some crowd developing.

## Motivation and Goal
Vizbox was born as a first step towards a Refbox. I'm not sure if this will be doable, but I completely support the idea of using it as visualization tool and, why not?, as an official alternate HRI interface.

Further on, Vizbox would also be part of an official *Deus Ex Machina* interface, extending the continue rule (see below).

## Milestones and Vizbox Future
As of now, Vizbox use is advised but nor enforced. Due to possible network problems we would like to keep it that way, and to keep the competition as fair as possible, avoid granting scores for its use. Instead, we would love to have Vizbox as an intuitive and transparent way to visualize the competition, making teams using it  (and the whole league) to have a tactical advantage over the rest. 

Some features that come to my mind are:
- Enhanced experience for the audience
  - On-screen augmented reality markers
  - Live tracking of Main/Bonus goals (i.e. Scoring)
  - CCTV surveillance
    - Visualize many rooms at the same time
    - At some point we may need to isolate the arena
  - Live info about the participant team
  - Timers
- Built in data recording (specially important for HRI).
- Supervised A.I./Feature bypassing, reducing score but helping the robot to finish the task.
  - Plan modification
  - Injecting raw audio signals (bypass microphone)
  - Injecting text commands (bypassing ASR)
  - Remote control (bypassing navigation/manipulation)
  - Position override (bypass localization)

Although is undesirable to have robots being remote controlled and nowadays is strictly forbidden, it would help every now and then in terms of the show to let a team finish the demo although  points wouldn't be scored. This would only be possible if all teams share the same interface.

In addition, we have plans of providing robots with recordings of NL commands mixed with controlled amounts of noise to keep the competition fair. Vizbox would help a lot with it.

**Comment by cdussieux** [comment]
> You are free to do that and present that as HRI alternative!

Great, thanks :)




@kyordhel 
Well, thank you for such a detailed answer. I think we'll definitely explore this path.
Of course discussion on further contribution will be moved to VizBox repo.
I am closing this.
Thanks
